### DISCUSSION TITLE
How to identify the reason behind packet drop in wirelesshost -omnetpp+inet 4,5

### DISCUSSION THREAD

**Discussion by Abh4git** [omnetpp/omnetpp#1024] [Q&A]
Hi OMNET team,

I am using OMNET 6.0.1 and INET 4.5. I am having a UDP Packet being sent from
Accesspoint to WirelessHost. It reaches the host, but after that gets dropped. There is a warning saying Unknown protocol dropping packet. I further checked  including PCAP Recorders. In Wireshark it shows UDP Packet, but still it gets dropped. Packet Drop Visualizer shows reason 13. How can I get to know more about the reason and the root cause. How best to diagnose further and find out the real reason.
Any thoughts?
--log---
INFO (Rx)Network3.ES4.wlan[0].mac.rx: Received frame from PHY: (inet::Packet)ES2-video-app2-0 (568 B) (inet::SequenceChunk) length = 597 B
INFO (Dcf)Network3.ES4.wlan[0].mac.dcf: Processing lower frame: ES2-video-app2-0
INFO (Dcf)Network3.ES4.wlan[0].mac.dcf: Processing received frame ES2-video-app2-0 as recipient.
** Event #104  t=0.000228614069  Network3.ES4.wlan[0].llc (Ieee80211LlcLpd, id=929)  on ES2-video-app2-0 (inet::Packet, id=320)

INFO: Received (inet::Packet)ES2-video-app2-0 (540 B) (inet::SequenceChunk) length = 597 B from lower layer.
WARN: Unknown protocol, dropping packet
** Event #105  t=0.000238614069  Network3.ES4.wlan[0].mac.tx (Tx, id=941)  on selfmsg endIFS (omnetpp::cMessage, id=186)

DETAIL: Tx: endIfsTimer expired

Thanks
Regards
Abhilash

**Comment by levy** [comment]
The best approach would be to put breakpoints where this error message could be printed and debug the simulation. My guess is that a protocol module is missing from the wireless network node, so when the packet gets decapsulated it cannot be dispatched to the desired protocol module.

**Comment by Abh4git** [comment]
Thank you, I tried this and found the issue is - I see 'Unknown protocol' warning is from this method
void Ieee80211LlcEpd::handleMessage(cMessage *message)
In pcap /wireshark I am seeing UDP packet, but at llc wireless is not able to recognize it.
Only difference is my packet is a TSN Device generated UDP Packet (maybe more headers related to 802.1Q). I will continue investigating, but any further thoughts or something I am missing?
